package.json

```diff
@@ -68,13 +68,13 @@
     "enquirer": "^2.3.4",
     "eslint": "^6.1.0",
     "eslint-config-prettier": "^6.0.0",
-    "eslint-config-react-app": "^5.0.2",
+    "eslint-config-react-app": "^5.2.1",
     "eslint-plugin-flowtype": "^3.13.0",
     "eslint-plugin-import": "^2.18.2",
     "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-prettier": "^3.1.0",
     "eslint-plugin-react": "^7.14.3",
-    "eslint-plugin-react-hooks": "^2.2.0",
+    "eslint-plugin-react-hooks": "^2.5.0",
```

Comment:
Did this actually need changing? Because I don't believe it does -- `^2.2.0` is compatible with `2.5.0`

On the same note, `eslint-config-react-app` shouldn't need changing either but it does seem to install `5.0.2` by default for some reason 🤔

Reply:
I updated the pull request as you suggested, but think we were both getting confused. We need to distinguish what happens when a `tsdx` developer runs `yarn install` in the `tsdx` repository from the end user case.
The warning appears when _developing_ tsdx itself, because the `yarn.lock` in the repository pins `eslint-config-react-app` to `5.0.2`. 
If a _user_ installs `tsdx` for the first time, they won’t get any error, as it will resolve the `eslint-config-react-app` to the latest compatible version, right now `5.2.1`.
The user _will_ see an error message if they run ` yarn add tsdx` when a new version comes out and they first installed `tsdx` when `eslint-config-react-app` was still at `5.0.2`, because in that case their own  `yarn.lock` would pin `eslint-config-react-app` (I tested this by setting the `eslint-config-react-app` dependency to exactly `5.0.2`, installing, then setting it to `^5.0.2`, installing again, then finally setting it to `^5.2.1` and installing a third time).
In the end, the minimal solution is still to do just as you suggest, and update only `eslint-config-react-app` to the latest version both in `package.json` and `yarn.lock`.
1. tsdx `yarn.lock` update (`yarn upgrade eslint-config-react-app` in tsdx repository) takes care of the warning while developing tsdx itself
2. For an upgrading user, `tsdx`’s own `yarn.lock` has no effect, so getting rid of the warning requires to also update `eslint-config-react-app` in `package.json`, otherwise the installer will keep using the version set in their own lockfile
3. Updating `eslint-plugin-react-hooks` is unnecessary because `2.2.0` is already compatible with the latest `eslint-config-react-app`

Reply:
> The warning appears when _developing_ tsdx itself, because the `yarn.lock` in the repository pins `eslint-config-react-app` to `5.0.2`.

Yea, was aware of that one, but that's not the one I meant. This one's not particularly important as it's not visible to end users.

> The user _will_ see an error message if they run ` yarn add tsdx` when a new version comes out and they first installed `tsdx` when `eslint-config-react-app` was still at `5.0.2`, because in that case their own `yarn.lock` would pin `eslint-config-react-app` (I tested this by setting the `eslint-config-react-app` dependency to exactly `5.0.2`, installing, then setting it to `^5.0.2`, installing again, then finally setting it to `^5.2.1` and installing a third time).

Ah that would make sense that their lockfile has the transitive dep set. What I was referencing actually was that I was getting this error when I was installing _globally_. I guess there's a global lockfile somewhere too that makes this happen 

> 2\. so getting rid of the warning requires to also update `eslint-config-react-app` in `package.json`, otherwise the installer will keep using the version set in their own lockfile

👍  . I thought this might be relevant to vulnerabilities too like https://github.com/formium/tsdx/pull/731#issuecomment-688246341, but an `npm audit fix` will automatically resolve those lockfile updates, whereas a warning like this one needs manual intervention

Great analysis @ludofischer !